File: 3471-find-the-largest-almost-missing-integer/3471-find-the-largest-almost-missing-integer.py
class Solution(object):
    def largestInteger(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        mp=Counter()
        freq=Counter()
        n=len(nums)
        r=0
        l=0
        while(r<n):
            mp[nums[r]]+=1
            if(r-l+1==k):
                for i in mp.items():
                    freq[i[0]]+=1
                mp[nums[l]]-=1
                if(mp[nums[l]]==0):
                    del mp[nums[l]]
                l+=1
            r+=1
        
        maxx=-1

        for i in freq.items():
            if(i[1]==1):
                maxx=max(maxx,i[0])
        return maxx

File: 3471-find-the-largest-almost-missing-integer/3471-find-the-largest-almost-missing-integer.cpp
class Solution {
public:
    int largestInteger(vector<int>& nums, int k) {
        int n=nums.size();
        unordered_map<int,int> mp;
        unordered_map<int,int> freq;
        int r=0;
        int l=0;
        while(r<n){
            mp[nums[r]]++;
            if(r-l+1==k){
                for(auto it:mp){
                    freq[it.first]++;
                }
                mp[nums[l]]--;
                if(mp[nums[l]]==0){
                    mp.erase(nums[l]);
                }
                l++;
            }
            r++;
        }
        int maxx=-1;
        for(auto it:freq){
            if(it.second==1){
                maxx=max(maxx,it.first);
            }
        }
        return maxx;
       
        
        
    }
};